src/App.tsx

```diff
@@ -1,73 +1,125 @@
 import React from 'react';
+import debounce from 'lodash.debounce';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
+
+export const Dropdown: React.FC<{
+  people: Person[];
+  onSelected: (person: Person) => void;
+}> = ({ people, onSelected }) => {
+  return (
+    <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+      <div className="dropdown-content">
+        {people.map(person => (
+          <div
+            key={person.slug}
+            className="dropdown-item"
+            data-cy="suggestion-item"
+          >
+            <p
+              onClick={() => onSelected(person)}
+              className={
+                person.fatherName && person.motherName
+                  ? 'has-text-link'
+                  : 'has-text-danger'
+              }
+            >
+              {person.name}
+            </p>
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [isActive, setIsActive] = React.useState(false);
+  const [people, setPeople] = React.useState(peopleFromServer);
+  const [selectedPerson, setSelectedPerson] = React.useState<Person | null>(
+    null,
+  );
+
+  const handleSearch = React.useCallback((searchValue: string) => {
+    const filteredPeople = peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(searchValue.toLowerCase()),
+    );
+
+    if (filteredPeople.length === 0) {
+      setIsActive(false);
+    } else {
+      setIsActive(true);
+    }
+
+    setSelectedPerson(null);
+    setPeople(filteredPeople);
+  }, []);
+
+  const debouncedSearch = React.useMemo(
+    () => debounce(handleSearch, 300),
+    [handleSearch],
+  );
+
+  React.useEffect(() => {
+    return () => {
+      debouncedSearch.cancel();
+    };
+  }, [debouncedSearch]);
```

Comment:
The `useEffect` hook for cleaning up the debounced function should include `debouncedSearch` in its dependency array. This ensures that the cleanup function is correctly updated whenever `debouncedSearch` changes.